flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/writer/BaseWriter.java

```diff
@@ -90,6 +92,12 @@ abstract class BaseWriter<IN> implements SinkWriter<IN> {
     // the records in a stream get committed to the table. Hence, records written to BigQuery
     // table is equal to this "totalRecordsWritten" only upon checkpoint completion.
     long totalRecordsWritten;
```

Comment:
@jayehwhyehentee As per offline discussion `totalRecordsWritten` now represents records written to BigQuery in both cases right? Since EXO would contain another metric `numberOfRecordsBufferedByBigQuerySinceCheckpoint` which would contain the count of number of records appended but yet remain to be committed to the table. The current `totalRecordsWritten`  as per the comment would represent the total records buffered by BigQuery which metric is not useful as such to the users.

Reply:
```
    // In exactly-once mode, "totalRecordsWritten" actually represents records appended to a
    // write stream by this writer. Only at a checkpoint, when sink's commit is invoked, will
    // the records in a stream get committed to the table. Hence, records written to BigQuery
    // table is equal to this "totalRecordsWritten" only upon checkpoint completion.
    long totalRecordsWritten;
```
As mentioned in this comment, for EXO, `totalRecordsWritten` is the actual number of records written (not buffered) to BigQuery table only for the duration between committer.commit (after checkpoint) and first call to writer.write